agent/tool-scripts/kvm-spinlock

```diff
@@ -258,42 +258,45 @@ function safe_kill() {
 		# should not happen
 		return 1
 	fi
-	
+
 	# check that the pid corresponds to the tool
 	pidlist=$(pidof -x $tool)
-	typeset -i len=0
 	for p in $pidlist ;do
-		len=$len+1
 		if [[ $p == $pid ]] ;then
 			pid_to_kill=$pid
 		fi
 	done
-	# hack: individual tools will have a different number of pids running.
-	# we might just delete the warning if it gets too cumbersome.
-	if [[ $len > 4 || ($len > 3 && $tool != "turbostat") ]] ;then
-		warn_log "Too many pids for $tool: $pidlist -- maybe old tools running? Use pbench-kill-tools."
-	fi
-	
+
 	rc=0
 	if [ ! -z "$pid_to_kill" ] ;then
 		kill -s $signal $pid_to_kill 2>/dev/null
 		rc=$?
 	fi
-	
 	[[ $rc == 0 ]] && return 0
 
 	# check if the process is still running.
+	pid_to_kill=""
+	typeset -i len=0
 	pidlist=$(pidof -x $tool)
 	for p in $pidlist ;do
+		len=$len+1
 		if [[ $p == $pid ]] ;then
 			# why is the pid still there?
-			kill -s KILL $pid 2>/dev/null
-			return 2
+			pid_to_kill=$pid
 		# else
 			# why are there other tool processes running?
 			# should we kill them?
 		fi
 	done
+
+	if [ ! -z "$pid_to_kill" ] ;then
+		kill -s KILL $pid_to_kill 2>/dev/null
+		return 2
+	fi
+
+	if [[ $len > 4 || ($len > 3 && $tool != "turbostat") ]] ;then
```

Comment:
Should there be a comment here that explain why there is a special case for turbostat?

Reply:
Yes there should be.  Does anybody know the history of that check?

Reply:
@atheurer ?

Reply:
@ndokos, it looks like the `turbostat` check landed in PR #199, back in v0.39 of the pbench-agent.

Do you remember why we need to check for `turbostat`?

Reply:
Sorry - I don't remember.  I'll try to instrument safer_kill and see why turbostat seems to behave differently from everything else.

Reply:
I get 3 pids from `pidof -x turbostat` and also 3 pids from `pidof -x iostat`. The only difference is that there is a turbostat-datalog between the pbench-tool turbostat call and the "real" one, but pidof does not count  the datalog process.  So far, there does not seem to be a reason for treating turbostat differently.

Reply:
Okay, can we track this as a separate piece of work in issue #887, and not make further changes here, since this PR did not change that behavior?

Reply:
Since I've already approved, I'll wait for an OK from either @k-rister or/and @theurer and merge it.